Dockerfile

```
@@ -1,32 +1,36 @@
-ARG ROS_VERSION=foxy
-FROM ghcr.io/aica-technology/ros2-control-libraries:${ROS_VERSION} as core-dependencies
+FROM ghcr.io/epfl-lasa/control-libraries/development-dependencies as source-dependencies
 
-RUN sudo apt-get update && sudo apt-get install -y \
-  libmbedtls-dev \
-  libsodium-dev \
-  libzmq3-dev \
-  && sudo rm -rf /var/lib/apt/lists/*
+RUN apt-get update && apt-get install -y libmbedtls-dev libsodium-dev libzmq3-dev \
+  && apt-get clean && rm -rf /var/lib/apt/lists/*
 
-# install cppzmq bindings
-ARG CPPZMQ_VERSION=4.7.1
 WORKDIR /tmp
+ARG CPPZMQ_VERSION=4.7.1
 RUN wget https://github.com/zeromq/cppzmq/archive/v${CPPZMQ_VERSION}.tar.gz -O cppzmq-${CPPZMQ_VERSION}.tar.gz
 RUN tar -xzf cppzmq-${CPPZMQ_VERSION}.tar.gz
 WORKDIR /tmp/cppzmq-${CPPZMQ_VERSION}
 RUN mkdir build && cd build && cmake .. -DCPPZMQ_BUILD_TESTS=OFF && make -j install
+
 WORKDIR /tmp
-RUN rm -rf cppzmq*
+ARG CONTROL_LIBRARIES_BRANCH=develop
+RUN git clone -b ${CONTROL_LIBRARIES_BRANCH} --single-branch https://github.com/epfl-lasa/control-libraries.git
```

```suggestion
RUN git clone -b ${CONTROL_LIBRARIES_BRANCH} --depth 1 https://github.com/epfl-lasa/control-libraries.git
```

`--depth 1` is a bit stronger than `--single-branch` as it also pull only the latest commit. This makes things even lighter which is good here. There is no need for both as depth assumes single branch.